src/main/kotlin/org/gradle/script/lang/kotlin/support/ImplicitImports.kt

```diff
@@ -16,11 +16,10 @@
 
 package org.gradle.script.lang.kotlin.support
 
-object ImplicitImports {
-    val list = listOf(
-        "org.gradle.api.*",
-        "org.gradle.api.plugins.*",
-        "org.gradle.script.lang.kotlin.*",
+import org.gradle.configuration.ImportsReader
 
-        "java.io.File")
-}
+class ImplicitImports(
+    importsReader: ImportsReader,
+    val list: List<String> = importsReader.importPackages.map { "$it.*" }.plus(listOf(
```

Comment:
I'd keep the list of imports as a lazy field and introduce two explaining functions.

```
class ImplicitImports(private val importsReader: ImportsReader) {

    val list by lazy {
        gradleImports() + gradleScriptKotlinImports()
    }

    fun gradleImports() = importsReader.importPackages.map { "$it.*" }

    fun gradleScriptKotlinImports() = listOf("org.gradle.script.lang.kotlin.*", "java.io.File")
}
```